osquery/tables/networking/curl_certificate.cpp

```diff
@@ -335,49 +352,133 @@ Status getTLSCertificate(const std::string& hostname,
     return Status::failure("Failed to create OpenSSL CTX object");
   }
 
-  auto delBIO = [](BIO* bio) { BIO_free_all(bio); };
-  auto server = std::unique_ptr<BIO, decltype(delBIO)>(
-      BIO_new_ssl_connect(ctx.get()), delBIO);
-  if (server == nullptr) {
-    return Status::failure("Failed to create OpenSSL BIO object");
+  std::string port = "443";
+  auto connect_hostname = hostname;
+  auto delim = hostname.find(":");
+  if (delim + 1 == hostname.length()) {
+    // if no port specified use default port
+    connect_hostname = hostname.substr(0, delim);
+  } else if (delim != std::string::npos) {
+    port = hostname.substr(delim + 1, std::string::npos);
+    connect_hostname = hostname.substr(0, delim);
+  }
+
+  int ret;
+#ifdef WIN32
+  // Initialize Winsock
+  WSADATA wsaData;
+  ret = WSAStartup(MAKEWORD(2, 2), &wsaData);
```

Comment:
I see you calling `WSACleanup()` before returning below, checkout our `./osquery/utils/scope_guard.h` utilities
```cpp
auto const wsa_guard = scope_guard::create([]() { WSACleanup(); });
```

Reply:
Then you can remove all of the `#ifdef WIN32` + `WSAClean()` + `#endif` calls below.

Reply:
i think i did this right, but i'd appreciate a look